sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/src/SchemaRegistryAvroEncoder.cs

```diff
@@ -58,19 +54,114 @@ private enum SupportedType
             GenericRecord
         }
 
-        private static SupportedType GetSupportedTypeOrThrow(Type type)
+        #region Encode
+
+        /// <summary>
+        /// Encodes the message data into Avro and stores it in <see cref="MessageWithMetadata.Data"/>. The <see cref="MessageWithMetadata.ContentType"/>
+        /// will be set to "avro/binary+schemaId" where schemaId is the ID of the schema used to encode the data.
+        /// </summary>
+        /// <param name="data">The data to serialize to Avro and encode into the message.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        /// <typeparam name="TMessage">The <see cref="MessageWithMetadata"/> type to encode the data into.</typeparam>
```

Comment:
nit:  This read a bit awkward to me.  Should we consider normalizing with the `messageType` parameter description used below?  I believe they represent the same thing.

```suggestion
        /// <typeparam name="TMessage">The type of message to encode the data as. Must extend from <see cref="MessageWithMetadata"/> and have a parameterless constructor.</typeparam>
```